### DISCUSSION TITLE
正文参考文献上标显示问号

### DISCUSSION THREAD

**Discussion by zilangch** [note286/xduts#74] [Q&A] [answered]
目录结构
└─paper
    │  paper.tex
    │
    ├─chapters
    │  └─chapter-1
    │          chapter-1.tex
    │
    └─reference
            reference.bib

![Uploading image.png…]()
![Uploading image.png…]()

重新编译运行了几次结果还是一样的

**Comment by zilangch** [comment]
正文：研究热点[?]
latex：研究热点\cite{b2}   ps: Citation missing: b2

**Comment by note286** [comment]
首先，请等待图片上传完成后再提交。其次，请认真阅读手册中关于编译的说明。

![image](https://user-images.githubusercontent.com/16207292/225653241-70acdad0-de99-4b00-9b06-d9c9e563cbe3.png)

**Comment by zilangch** [comment]
上面的的问题是在  bib-backend = bibtex, 的情况下，上角标显示[?]
如果 bib-backend = biblatex, biblatex-option = { gbnamefmt = quanpin }, 上角标显示 b2

**Comment by note286** [comment]
我完全不知道你到底做了什么操作，例如，`.bib`文件是否正确配置，即`info/bib-resource`；还有，编译命令是否使用正确，这与`style/bib-backend`有关。最后，你可能并不知道`gbnamefmt=quanpin`是什么意思，不知道就不要用。

**Comment by zilangch** [comment]
我没找到问题在哪😂
![99A79329-DECF-4847-B798-D94660B73201](https://user-images.githubusercontent.com/28552847/225659926-a268e053-b065-4349-b747-fec854886004.png)
![42A5EE95-07F0-4FFB-9698-190204859829](https://user-images.githubusercontent.com/28552847/225659944-49c5a625-f3a4-4bb8-9da5-b5ac5b4aaf7f.png)
![425B5FB9-3F3B-4F77-9025-048B1F1871B6](https://user-images.githubusercontent.com/28552847/225659953-c9f637a8-58b2-47ab-a277-b07a857d4584.png)
![9CFE493B-309E-48F1-A873-5D7DEE41910B](https://user-images.githubusercontent.com/28552847/225659966-66dc48cd-cc77-4df1-b0f5-91a5391099e4.png)
![5EF80352-CB77-48D8-BA89-4DF7299EA6FE](https://user-images.githubusercontent.com/28552847/225659978-d13092fa-875e-4b34-9551-26a1624a6995.png)

**Reply by note286** [reply]
你觉得我能根据目前你所给的信息复现你的操作吗？

**Reply by note286** [reply]
不知道你写论文的时候是否也这样只言片语，你要清晰地描述，okay？

**Reply by zilangch** [reply]
emm，这个的日志您需要的话我可以截图😂，但是我不太会看这个日志
![26A9F2B4-2493-41DB-9151-935553988391](https://user-images.githubusercontent.com/28552847/225661225-efe88fa6-b55b-4691-b259-c6a59dd99bf6.png)

**Reply by note286** [reply]
至今你一楼的描述没有任何价值，目录结构完全看不清楚到底是什么结构？两张图片完全没有上传成功。

**Reply by note286** [reply]
打上次你提问论文封面我就发现和你沟通起来很困难。

想要正常使用参考文献，请配置好`.bib`文件并使用相应的命令编译即可。

**Reply by zilangch** [reply]
一楼的图都在四楼补充了，您看您需要啥
![48CAB3A4-A0FA-44F5-B0E4-C2BD2391FC16](https://user-images.githubusercontent.com/28552847/225662477-7862c020-fbba-4b1a-aff2-101e90c42425.png)

**Reply by note286** [reply]
> 目录结构完全看不清楚到底是什么结构？

麻烦你认真一点，你从来不认真阅读完整我的回复。

**Reply by zilangch** [reply]
目录结构主要如图
![FA6CA56A-12FE-4DEE-A9CE-047F07499E75](https://user-images.githubusercontent.com/28552847/225665263-f9f7c3ed-d0e8-4b05-b5b6-5ee1e8155e00.png)
![EE1BDE19-947C-4283-876C-02559B50EE5B](https://user-images.githubusercontent.com/28552847/225665280-66c66ff1-9449-4b53-870e-556fe9d0c75d.png)

编译命令
![83C93035-CFD7-40A8-8937-8DEA4F7FDFD2](https://user-images.githubusercontent.com/28552847/225665987-8488479e-3a09-423d-b0c7-f01864b9dd71.png)
![F1E8E3BC-19ED-406E-AD52-3BCF07D4A02E](https://user-images.githubusercontent.com/28552847/225666016-3594c292-e997-4097-a811-06dcb90472a7.png)

**Comment by note286** [comment, marked as answer]
我给你一个示例，已经很清楚了，如果还不懂，你就有点耐心，认真点，多试试。

`main.tex`和`ref.bib`在同一目录下。

```latex
\documentclass{xdupgthesis}
\xdusetup{info/bib-resource=ref.bib}
\begin{document}
\cite{xxx}\cite{yyy}
\end{document}
```

```
@proceedings{xxx,
  editor    = {牛志明 and 斯温兰德 and 雷光春},
  title     = {综合湿地管理国际研讨会论文集},
  address   = {北京},
  publisher = {海洋出版社},
  year      = {2012},
}
@book{yyy,
  author    = {Fan, Xuetong and Sommers, Christopher H.},
  title     = {Food Irradiation Research and Technology},
  edition   = {2},
  address   = {Ames, Iowa},
  publisher = {Blackwell Publishing},
  year      = {2013},
  pages     = {25--26},
}
```

编译命令为：

```
xelatex -synctex=-1 main
biber main
xelatex -synctex=-1 main
```

![image](https://user-images.githubusercontent.com/16207292/225665549-9d6346b9-3af7-43fd-b590-e9d274f8583d.png)
![image](https://user-images.githubusercontent.com/16207292/225665827-d69ded28-0715-4c0f-b02a-2f36ac1342fd.png)

**Comment by sikouhjw** [comment]
文件目录的显示，可以参考
![图片](https://user-images.githubusercontent.com/49770925/225666729-60323e8e-2f30-46ba-9d40-668626398e35.png)
```tex
\dirtree{%
  .1 /.
  .2 cover-background\DTcomment{背景图}.
  .3 doctor.jpeg\DTcomment{博士背景图}.
  .3 master.jpeg\DTcomment{学硕背景图}.
  .3 promaster.jpeg\DTcomment{专硕背景图}.
  .2 user-manual\DTcomment{用户手册}.
  .3 fdudoc.cls.
  .3 gdutthesis.dtx\DTcomment{用户手册源码}.
  .2 .gitattributes.
  .2 .gitignore.
  .2 gdutthesis.cls\DTcomment{模板文档类}.
  .2 gdutthesis-template.bib\DTcomment{参考文献数据库}.
  .2 LICENSE.
  .2 main.tex\DTcomment{空白模板}.
  .2 README.md\DTcomment{简要自述}.
}
```

修正你的认知错误：
- 编译命令是指 `xelatex`, `biber` or `bibtex`
- 当文件在同一目录下，可以直接调用 `\input{xxx.tex}` or `\xdusetup{info/bib-resource=ref.bib}`，如果在子目录，可以 `\xdusetup{info/bib-resource={xxxx/ref.bib}}`
- 如果你给出的信息不足以让『不是你的第三者』认识到问题在哪，那就是你的信息不足，你需要补充信息，跟写论文没有区别
- 为什么要补充信息可以参考 https://zhuanlan.zhihu.com/p/607183745

**Reply by zilangch** [reply]
主要涉及到的目录
xduts-main
  GMSS.tex
  reference
    reference.bib
  chapaters
    chaper-1
      chapter-1.tex

主要代码
![image](https://user-images.githubusercontent.com/28552847/225674770-67cffa58-5f08-4678-a27a-23eee5d93c71.png)
![image](https://user-images.githubusercontent.com/28552847/225674891-f3ac01d8-9886-4b65-8d86-853a0d5f9b4f.png)

我感觉问题出在编译上
`xelatex -synctex=-1 main
biber main
xelatex -synctex=-1 main`
这段编译命名我并不了解以及不知道在哪运行

**Reply by zilangch** [reply]
主要涉及到的目录
xduts-main
##GMSS.tex
##reference
####reference.bib
##chapaters
####chaper-1
######chapter-1.tex

**Reply by zilangch** [reply]
‘main.tex和ref.bib在同一目录下  ’  还是一样的

**Reply by note286** [reply]
你找个会LaTeX的同学帮忙看看吧，GitHub讨论区救不了你，我都怀疑你压根没编译参考文献。你去看一下 https://github.com/note286/xduts/discussions/40#discussioncomment-4956807 中的回复，没准对你有帮助。

**Reply by zilangch** [reply]
好的，我看看去

**Reply by note286** [reply]
> 这段编译命名我并不了解以及不知道在哪运行

不要说出这种话，你都是要研究生毕业的人了，我给出了代码以及运行方式，并展示了运行结果。只要你不是怀疑我在骗你，你就应该去了解一下你不懂的内容。怎么了解？西安电子科技大学免费给你的校园网是干啥的。

**Reply by zilangch** [reply]
我有个问题，就是用texstudio的话，也需要去命令行里执行您列出的编译命令吗？根据您的回复，我猜测是不是我漏了这一步，之前用IEEE的模板的时候没有去命令行里执行命令

**Reply by note286** [reply]
你去看 https://github.com/note286/xduts/discussions/40#discussioncomment-4956807 吧，相信你会有所收获。

**Reply by note286** [reply]
> 这段编译命名我并不了解以及不知道在哪运行

另外，你既然会`tree`，我不相信你不知道怎么运行。

**Reply by sikouhjw** [reply]
> 我有个问题，就是用texstudio的话，也需要去命令行里执行您列出的编译命令吗？根据您的回复，我猜测是不是我漏了这一步，之前用IEEE的模板的时候没有去命令行里执行命令

1. 命令行编译只是帮助你『理解并掌握』编译的本质
2. texstudio 本质上就是调用了命令行编译
3. ieee 的模板，如果用到了 bibtex，那么也需要 `bibtex` 编译，如果没有『显式』用到，说明你的 texstudio 的配置默认为 `bibtex`，你需要将其修改为 `biber`

**Reply by zilangch** [reply]
问题已解决，多谢两位大佬的耐心解答同时也让我认识到了自己的不足，感恩！

环境：TexStudio
Default Bibliography Tool：biber
我自己的设置：
bib-backend = bibtex
在上面的环境和设置下导致了上述问题
解决：
将Default Bibliography Tool 设置为 bibtex

复现验证：
删除了相关的主tex文件（.bbl, .pdf等）
bib-backend = bibtex；Default Bibliography Tool：biber    复现出上述问题
bib-backend = bibtex；Default Bibliography Tool：bibtex    解决上述问题

**Reply by note286** [reply]
@zilangch 你可以去看一下[《一份（不太）简短的LaTeX2ε介绍》](http://mirrors.ctan.org/info/lshort/chinese/lshort-zh-cn.pdf)中的第一章LaTeX的基本概念。

**Reply by zilangch** [reply]
好的！
